vizro-core/src/vizro/managers/_model_manager.py

```diff
@@ -50,90 +50,58 @@ def __iter__(self) -> Generator[ModelID, None, None]:
 
         Note this yields model IDs rather key/value pairs to match the interface for a dictionary.
         """
+        # TODO: should this yield models rather than model IDs? Should model_manager be more like set with a special
+        #  lookup by model ID or more like dictionary?
         yield from self.__models
 
-    # TODO: Consider adding an option to iterate only through specific page - "in_page_with_id=None"
-    def _items_with_type(self, model_type: type[Model]) -> Generator[tuple[ModelID, Model], None, None]:
-        """Iterates through all models of type `model_type` (including subclasses)."""
-        for model_id in self:
-            if isinstance(self[model_id], model_type):
-                yield model_id, cast(Model, self[model_id])
-
-    # TODO: Consider returning with yield
-    # TODO: Make collection of model ids (throughout this file) to be set[ModelID].
-    def _get_model_children(self, model_id: ModelID, all_model_ids: Optional[list[ModelID]] = None) -> list[ModelID]:
-        if all_model_ids is None:
-            all_model_ids = []
-
-        all_model_ids.append(model_id)
-        model = self[model_id]
-        if hasattr(model, "components"):
-            for child_model in model.components:
-                self._get_model_children(child_model.id, all_model_ids)
-        if hasattr(model, "tabs"):
-            for child_model in model.tabs:
-                self._get_model_children(child_model.id, all_model_ids)
-        return all_model_ids
-
-    # TODO: Consider moving this method in the Dashboard model or some other util file
-    def _get_model_page_id(self, model_id: ModelID) -> ModelID:  # type: ignore[return]
+    def _get_models(
```

Comment:
Good idea! I'll make `_get_model_children` into `__get_model_children`.

As for making these methods public: I'm going to wait until @maxschulz-COL and I have reworked the model manager a bit before doing that because I think it's quite likely these will change again.

Reply:
[90d5868](https://github.com/mckinsey/vizro/pull/903/commits/90d586813fc33ffc28f272f1a27e7c7904b72a8d)